tests/test_utils.py

```diff
@@ -287,7 +288,7 @@ def test_no_positional_on_function():
     def t(foo=False):
         return foo
 
-    with pytest.raises(TypeError):
+    with pytest.raises(twine.exceptions.InvalidConfiguration):
```

Comment:
Based on which tests you're modifying in this file, I think you've taken the refactoring too far. `utils.no_positional` is internal only and should never show its face to the user of the CLI. A future user of the API might see a TypeError as a result but that's for a *very specific* reason of mimicing the behaviour of Python 3's keyword-only arguments. The message was very specifically chosen to mimic that as well.